scripts/write_height_dataset_to_bq.py

```diff
@@ -153,11 +153,14 @@ def main(args):
         # and add end of MDP rewards. in this problem all rewards are immediate
         # so add these as 0's to test the joining logic in training. in practice
         # end of MDP metrics' keys map to previous decisions
+        rand_previous_decision = random.choice(list(GENDER_MAP.values()))
```

Comment:
Hello, I am not sure to understand this, I would have thought that the decision would be the same as in the immediate reward. That is what seems to be explained in the comment just above. 

This is non breaking because delayed rewards are not considered after in the demo.

Reply:
So delayed rewards are logged once at the end of a session (or `mdp`) and have keys that point back to the decisions previously made across the `mdp`. For example, let's say you are doing product recommendations. In this case you'd have some immediate rewards (e.g. a `click`) and a delayed `purchase` reward.

Perhaps you recommend 5 products throughout the session - products A, B, C, D, & E. Let's say the user clicks on products C & E. This would fire off two immediate rewards for decision `C` and decision `E` with a `click` value of 1.

At the end of the session, perhaps a user buys product `A` and `E`. After this, we should fire off a delayed reward that looks like:

```
{"A": {"purchase": 1}, "E": {"purchase": 1}}
```

We will then assign these delayed rewards back to the decisions `A` and `E` across the session. Does that clear things up for you?

Reply:
Ah ok thanks for the very clear explanation!